ketchup/main.py

```diff
@@ -1,121 +1,231 @@
-
-    answer_today = input("Will you start watching today? [Y/n]: ") or "y"
-    if answer_today.lower() == "y":
-        start_date = dt.datetime.today()
-    else:
-        start_date_prompt = "What day will you start?(15 Aug 2023): "
-        start_date = get_date(start_date_prompt)
-
-    per_day_limit = (
-        input("What is the limit of episodes you will watch per day? [0+] ") or 0
-    )
-
-    # The following seems as though it could be better...
-    # We want to set the per day limit as an integer, or None...
-    try:
-        per_day_limit = int(per_day_limit)
-        if per_day_limit == 0:
-            per_day_limit = None
-    except ValueError:
-        per_day_limit = None
-
-    # Time to math!
-    watch_schedule = wsb.calculate_watch_schedule(
-        episodes, end_date, start_date, per_day_limit
-    )
-    print(f"Wow, that's only {watch_schedule['days']} days between today and then !!!")
-
-    if watch_schedule["is_possible"]:
-        print(
-            f"You will have to watch {watch_schedule['base_eps_per_day']} episodes per day "
-            "to complete the show on time!"
-        )
-        if watch_schedule["extra_episodes"]:
-            print(
-                f"You will have {watch_schedule['extra_episodes']} extra,"
-                "you can add an additional watch day or tack them on to one of your other days."
+            end_date_prompt = (
+                "What date would you like to have completed this series? "
+                "(Example: 15 Aug 2050): "
             )
-    else:
-        print("I'm sorry, but that limit and end date make it impossible!")
-        print("Don't worry, consider these options instead:")
-        print(
-            f"Option 1: You can watch {watch_schedule['base_eps_per_day']} episodes per day to catch up"
-        )
-        if watch_schedule["extra_episodes"]:
-            print(
-                f"You will have {watch_schedule['extra_episodes']} extra,"
-                "you can add an additional watch day or tack them on to one of your other days."
+
+            answer_today = input(
+                "Will you start watching today? [Y/N]: ") or "y"
+
+            if answer_today.lower() == "y":
+                start_date = dt.datetime.today()
+                episode_duration_mins = 24
+                # Set the default episode duration in minutes
+            else:
+                start_date_prompt = "What day will you start?" \
+                                    " (Ex: 15 Aug 2023): "
+                start_date = get_date(start_date_prompt)
+
+            end_date = get_date(end_date_prompt, require_future=True)
+
+            if end_date < start_date:
+                print("The end date cannot be before the start date. "
+                      "Please provide valid dates.\n")
+                continue
+
+            time_available = end_date - start_date
+            min_hours_needed = math.ceil(episodes * episode_duration_mins / 60)
```

Comment:
Maybe we add a utils.py function for determining if episodes * duration is less than 24 hours in a day. lol

Reply:
Unsure about the logic behind this, I'm having a difficult understanding the task which you wish to set.

Quickly, this is a simple function that I made, for how I am trying to perceive what it is you're commenting regarding.

`def is_less_than_24_hours(episodes, episode_duration_mins):
    min_hours_needed = math.ceil(episodes * episode_duration_mins / 60)
    return min_hours_needed < 24`
    
    Let me know what you think, and how this can be properly implemented.